.github/workflows/compile-deploy.yml

```diff
@@ -0,0 +1,54 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Build and Compile push
+
+on:
+  push:
+  pull_request:
```

Comment:
Do we really need a second build on pull request, when we already have push? This is mostly useful when we expect to see pull requests raised from forks.

Reply:
I was hoping

concurrency:
  group: compile-${{ github.workflow }}-${{ github.ref }} # Recommended if you intend to make multiple deployments in quick succession.
  cancel-in-progress: true
  
would deal with this but it does not, there should be other ways, as on pr is required for external runs, but we don't want it to trigger on commits etc.